### PR TITLE
Allow fluids to flow between claims of the same owner

### PR DESCRIPTION
**Pull request by FreeMonoid**
Context: https://korobi.vq.lc/network/spigot/channel/griefprevention/logs/2020/11/19/#L14

```
00:13:32 Dummycord <F​re​eM​on​oi​d> I was thinking it might be worth checking for claim owner instead of just whether it's the same claim, in case it's this uncommon case of same owner having two adjacent claims
00:26:19 RoboMWM hm
00:26:21 RoboMWM I c
00:26:44 RoboMWM should just get rid of it
00:26:55 RoboMWM pretty sure bigscary prolly removed only part of it due to performance issues
00:27:00 RoboMWM and didn't get it all
00:29:58 Dummycord <F​re​eM​on​oi​d> get rid of preventing flow in survival? idk, it's pretty easy way to grief someone with water or worse, lava (I see it everywhere on servers with WG claims cause it's super easy to abuse there) so I think it's pretty useful
00:30:44 RoboMWM wild -> claim isn't blocked is i
00:30:46 RoboMWM it
00:31:01 Dummycord <F​re​eM​on​oi​d> it is blocked
00:31:12 Dummycord <F​re​eM​on​oi​d> you can still mess up wilderness, but at least claims are intact
00:31:17 RoboMWM huh
00:31:25 RoboMWM I thought he claimed to remove that
00:31:33 Dummycord <F​re​eM​on​oi​d> ¯\_(ツ)_/¯
00:32:09 RoboMWM there's a lot of ways you could mess up someone, and usually at that point I advocate to make ur borders stronger/bigger
00:32:20 RoboMWM but eh
00:32:33 RoboMWM only thing that ppl complained about are pistons
00:32:38 RoboMWM so I guess it stays
00:33:12 RoboMWM can PR the same owner check if u wanna - just be aware of admin claims having null owner and that sort of stuff
```

Tested combinations:

combinations

Each 10x10 square is a claim, color denotes the owner: green is wilderness, red is Player 1, yellow is Player 2, black is admin claim, white squares inside claims are subdivisions; sea lantern is where the water source block is.

### CODE REVIEW

**Comment by Hazugg** [comment]
I ask the same, I would like that both lava or water fluids can flow normally within a claim, I have players who want to make their farms and cannot protect them for this reason since lava or water does not flow and for most it needs that this is done, it would be very good if I could add it in an update

**Comment by smplfy** [comment]
This would be super helpful. Please merge if possible.

**Comment by Allowsik** [comment]
This looks  great i don't get it why they didn't review it yet.

**Comment by FreeMonoid** [comment]
@RoboMWM (or maybe @Jikoo) can I ask you for a review of this older but still applicable PR, so it maybe can land in the next release?

**Review by Jikoo — COMMENTED** [review]
I think my one reservation is stuff like shop or display areas - someone filling their allotted subclaim with lava could grief adjacent decorations, or vice versa with restricted subclaims. Some of that is down to intelligent design for claim owners, but if we're introducing a new behavior, they can't really be expected to have planned around it. I guess GP could leave this to an addon, but it makes me a bit uncomfortable.

**Comment by FreeMonoid** [comment]
@Jikoo sorry for a late reply: I've added a check for subdivisions, it now behaves exactly as 16.18.1 with regard to them:



Does this address your concerns?

**Comment by Jikoo** [comment]
I also think a restricted subclaim should block flow from its parent because the list of people with build trust inside it has no relation to the list of people who have build trust in the claim as a whole. Otherwise yes, looks good!

**Comment by FreeMonoid** [comment]
@Jikoo started adding a check for the restricted subclaims and the code became a bit too hard to follow, so I've ended up refactoring it a bit.

Re-did the test with subclaims added to the mix:
fluid flow test
Similar to initial test: green is wilderness, red is Player 1, yellow is Player 2, black is admin claim, white squares inside claims are regular subdivisions; pink squares inside claims are restricted subdivisions; sea lantern is where the water source block is.

Let me know if have any feedback on this!

**Review by Jikoo — APPROVED** [review]
Looks excellent to me 👍 Sorry for asking for such a logical mess

**Comment by FreeMonoid** [comment]
No problem at all! I think the implementation we ended up with here is much cleaner than what I initially proposed!

**Review by RoboMWM — APPROVED** [review]
